File: common/src/main/java/in/dream/common/models/FlightModels.java
package in.dream.common.models;

import java.util.List;

import org.joda.time.DateTime;

import com.fasterxml.jackson.annotation.JsonIgnore;

public class FlightModels {
	public class FlightSearchResults{
		public AxisData axisData;
		public ItineraryResults itineraryResults;
	}
	public class AxisData{	
		public double priceMin;
		public double priceMax;
		public List<DateAxisMinMax> dateAxisMinMaxList;
	}
	public class DateAxisMinMax{
		public int durationMin;
		public int durationMax;
		public String departMin;
		public String departMax;
		public String departMinFullChart;
		public String departMaxFullChart;
		public String arrivalMin;
		public String arrivalMax;
		public String arrivalMinFullChart;
		public String arrivalMaxFullChart;
		
		@JsonIgnore
		public DateTime departMinAsDate;
		@JsonIgnore
		public DateTime departMaxAsDate;
		@JsonIgnore
		public DateTime arrivalMinAsDate;
		@JsonIgnore
		public DateTime arrivalMaxAsDate;
	}
	public class ItineraryResults{
		public List<TripDetail> trips;		
	}
	public class TripDetail{
		public int tripId;
		public double saleTotal;	
		public List<Slice> slices;	
		public String html;		
	}
	public class Slice{
		public int duration;
		public String departureTime;
		public String arrivalTimeInOrigin;
		public String arrivalTime;		
		public List<Segment> segments;
	}
	public class Segment{
		public int duration;
		public int connectionDuration;
		public String flightCarrier;
		public List<Leg> legs;
	}
	public class Leg{
		public int duration;
		public int connectionDuration;
		public String departTimeInOrigin;
		public String arrivalTimeInOrigin;
		public String origin;
		public String destination;		
	}
}

File: frontend/web/src/ts/models/FlightModels.ts
module Flights {
	export class FlightSearchResults{
		public axisData: AxisData;
		public itineraryResults: ItineraryResults;
	}
	export class AxisData{	
		public priceMin: number;
		public priceMax: number;
		public dateAxisMinMaxList: DateAxisMinMax[];
		constructor (){}
	}
	export class DateAxisMinMax{
		constructor (
			public durationMin: number,
			public durationMax: number,
			public departMin: string,
			public departMax: string,
			public departMinFullChart: string,
			public departMaxFullChart: string,
			public arrivalMin: string,
			public arrivalMax: string,
			public arrivalMinFullChart: string,
			public arrivalMaxFullChart: string
		){}		
	}
	export class ItineraryResults{
		public trips: Array<TripDetail>;
		constructor (){}
	}
	export class TripDetail{
		public tripId: number;
		public saleTotal: number;
		public slices: Array<Slice>;
		public html: string;		
		constructor (){}	
	}	
	export class Slice{
		public duration: number;
		public departureTime: string;
		public arrivalTimeInOrigin: string;
		public arrivalTime: string;
		public segments: Array<Segment>;
		constructor(){}
	}
	export class Segment{
		public duration: number;
		public connectionDuration: number;
		public flightCarrier: string;
		public legs: Array<Leg>;
		constructor(){}
	}
	export class Leg{
		public duration: number;
		public connectionDuration: number;
		public departTimeInOrigin: string;
		public arrivalTimeInOrigin: string;
		public origin: string;
		public destination: string;		
		constructor(){}
	}
}